Description: Fix GPG signature validation on Python 3.
Origin: commit, revision id: jelmer@jelmer.uk-20190113142418-u7cljkak9krox8uk
Author: Jelmer VernooÄ³ <jelmer@jelmer.uk>
Last-Update: 2019-01-13
Applied-Upstream: no
X-Bzr-Revision-Id: jelmer@jelmer.uk-20190113142418-u7cljkak9krox8uk

=== modified file 'breezy/gpg.py'
--- old/breezy/gpg.py	2018-11-19 01:38:53 +0000
+++ new/breezy/gpg.py	2019-01-13 14:24:18 +0000
@@ -313,11 +315,12 @@
         if result.signatures[0].summary & gpg.constants.SIGSUM_VALID:
             key = self.context.get_key(fingerprint)
             name = key.uids[0].name
+            if isinstance(name, bytes):
+                name = name.decode('utf-8')
             email = key.uids[0].email
-            return (
-                SIGNATURE_VALID,
-                name.decode('utf-8') + u" <" + email.decode('utf-8') + u">",
-                plain_output)
+            if isinstance(email, bytes):
+                email = email.decode('utf-8')
+            return (SIGNATURE_VALID, name + u" <" + email + u">", plain_output)
         # Sigsum_red indicates a problem, unfortunatly I have not been able
         # to write any tests which actually set this.
         if result.signatures[0].summary & gpg.constants.SIGSUM_RED:

